In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from dateutil.parser import parse
%matplotlib inline

In [9]:
df=pd.read_csv('apr11.csv',
               parse_dates=['dateRep']
              )
df.head()

dateRep  day  month  year  cases  deaths countriesAndTerritories geoId  \
0 2020-10-04   10      4  2020     61       1             Afghanistan    AF   
1 2020-09-04    9      4  2020     56       3             Afghanistan    AF   
2 2020-08-04    8      4  2020     30       4             Afghanistan    AF   
3 2020-07-04    7      4  2020     38       0             Afghanistan    AF   
4 2020-06-04    6      4  2020     29       2             Afghanistan    AF   

  countryterritoryCode  popData2018  
0                  AFG   37172386.0  
1                  AFG   37172386.0  
2                  AFG   37172386.0  
3                  AFG   37172386.0  
4                  AFG   37172386.0

# Don't just dive in...
- explore first :)

In [12]:
# using value_counts() on one column
df['countriesAndTerritories'].value_counts()

Spain                          102
Iceland                        102
Norway                         102
South_Korea                    102
China                          102
                              ... 
Malawi                           8
Falkland_Islands_(Malvinas)      7
South_Sudan                      5
Sao_Tome_and_Principe            2
Yemen                            1
Name: countriesAndTerritories, Length: 206, dtype: int64

### ---> There are 206 countries and territories in the dataset

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9922 entries, 0 to 9921
Data columns (total 10 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   dateRep                  9922 non-null   datetime64[ns]
 1   day                      9922 non-null   int64         
 2   month                    9922 non-null   int64         
 3   year                     9922 non-null   int64         
 4   cases                    9922 non-null   int64         
 5   deaths                   9922 non-null   int64         
 6   countriesAndTerritories  9922 non-null   object        
 7   geoId                    9895 non-null   object        
 8   countryterritoryCode     9725 non-null   object        
 9   popData2018              9769 non-null   float64       
dtypes: datetime64[ns](1), float64(1), int64(5), object(3)
memory usage: 775.3+ KB


### There are 10 columns in total, with data types: datetime64, int64, object (string), float64

In [14]:
df.describe()

day        month         year         cases       deaths  \
count  9922.000000  9922.000000  9922.000000   9922.000000  9922.000000   
mean     15.299536     2.649869  2019.993247    157.615098     9.579117   
std       9.309211     1.288910     0.081901   1226.244581    78.833194   
min       1.000000     1.000000  2019.000000     -9.000000     0.000000   
25%       7.000000     2.000000  2020.000000      0.000000     0.000000   
50%      15.000000     3.000000  2020.000000      0.000000     0.000000   
75%      24.000000     3.000000  2020.000000     15.000000     0.000000   
max      31.000000    12.000000  2020.000000  34272.000000  2004.000000   

        popData2018  
count  9.769000e+03  
mean   6.322340e+07  
std    1.995458e+08  
min    1.000000e+03  
25%    3.323929e+06  
50%    1.028176e+07  
75%    4.222843e+07  
max    1.392730e+09

### This shows the statistical distribution of each column (attribute). But only SOME are USEFUL.
- On average, there are 158 new cases and 10 new deaths by covid-19 every day in the world.

In [19]:
df.dtypes

dateRep                    datetime64[ns]
day                                 int64
month                               int64
year                                int64
cases                               int64
deaths                              int64
countriesAndTerritories            object
geoId                              object
countryterritoryCode               object
popData2018                       float64
dtype: object

### df.dtypes method is not very useful because it has been covered in df.info()